RecoLocalTracker/Phase2TrackerRecHits/plugins/Phase2TrackerRecHits.cc

```diff
@@ -77,11 +77,11 @@ void Phase2TrackerRecHits::produce(edm::StreamID sid, edm::Event& event, const e
     // Container for the clusters that will be produced for this modules
     Phase2TrackerRecHit1DCollectionNew::FastFiller rechits(*outputRecHits, DSViter.detId());
 
-    for (auto clustIt : DSViter) { 
-      ClusterParameterEstimator< Phase2TrackerCluster1D >::LocalValues lv = cpe->localParameters(clustIt, *geomDetUnit);
+    for (edmNew::DetSet<Phase2TrackerCluster1D>::const_iterator clustIt = DSViter.begin(); clustIt != DSViter.end(); ++clustIt) { 
```

Comment:
It would seem safer to make line 70:
```
 for (const auto &DSViter : *clusters) { 
```
and line 80:
```
for (const auto &clustIt : DSViter) { 
```
so there's no copying, and then `&clustIt` would point back into `clusters`, so the original versions of line 81 and 84 could be used.
In this case, the names of `DSViter` and `clustIt` should be changed, because neither of them are iterators. `DSViter` could be called `clusterDetSet`, and `clustIt` could be called `clusterRef`.